src/components/ObsDetails/ObsDetailsContainer.js

```diff
@@ -121,6 +129,7 @@ const ObsDetailsContainer = ( ): Node => {
   const { params } = useRoute();
   const {
     suggestedTaxonId,
+    taxon,
```

Comment:
One issue I'm seeing with passing this full taxon object through params is the following warning. If we're going to use params, I'd say we should only use the parts of the taxon object we need to show the user or send to the API (i.e. taxon id, photo, name, common name). 

```
 WARN  Non-serializable values were found in the navigation state. Check:

TabNavigator > TabStackNavigator > ObsDetails > params.taxon.default_photo._synced_at (Tue Jul 02 2024 08:57:56 GMT-0700)

This can break usage such as persisting and restoring state. This might happen if you passed non-serializable values such as function, class instances etc. in params. If you need to use components with callbacks in your options, you can use 'navigation.setOptions' instead. See https://reactnavigation.org/docs/troubleshooting#i-get-the-warning-non-serializable-values-were-found-in-the-navigation-state for more details.
```